lib/connectors/memory.js

```diff
@@ -723,6 +723,67 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replaceById = function(model, id, data, options, cb) {
+  var self = this;
+  if (!id) {
+    var err = new Error('You must provide an id when replacing!');
+    return cb(err);
+  }
+  // Do not modify the data object passed in arguments
+  data = Object.create(data);
+  this.setIdValue(model, data, id);
+  var cachedModels = this.collection(model);
+  var modelData = cachedModels && this.collection(model)[id];
+  if (!modelData) {
+    return cb(new Error('Could not replace. Object with id ' + id + ' does not exist!'));
+  }
+
+  var newModelData = {};
+  for(var key in data) {
+    var val = data[key];
+    if(typeof val === 'function') {
+      continue; // Skip methods
+    }
+    newModelData[key] = val;
+  }
+
+  this.collection(model)[id] = serialize(newModelData);
+  this.saveToFile(newModelData, function (err) {
+    cb(err, self.fromDb(model, newModelData));
+  });
+};
+
+Memory.prototype.replaceOrCreate = function(model, data, options, callback) {
+  var self = this;
+  var idName = self.idNames(model)[0];
+  var idValue = self.getIdValue(model, data);
+  var filter = {where: {}};
+  filter.where[idName] = idValue;
+  var nodes = self._findAllSkippingIncludes(model, filter);
+  var found = nodes[0];
+
+  if (!found) {
+    // Calling _createSync to update the collection in a sync way and 
+    // to guarantee to create it in the same turn of even loop
+    return self._createSync(model, data, function(err, id) {
+      if (err) {
+        return process.nextTick(function() {
+          cb(err);
+        });
+      };
```

Comment:
Would you mind compressing this block to a single line?

```js
if (err) return process.nextTick(function() { cb(err); });
```

Or at least

```js
if (err) {
  return process.nextTick(function() { cb(err); });
}
```

The current version has too low signal/noise ratio for me.